projects/plugins/boost/app/modules/cache/Page_Cache_Setup.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+namespace Automattic\Jetpack_Boost\Modules\Page_Cache;
+
+class Page_Cache_Setup {
+
+	/**
+	 * Runs setup steps and returns whether setup was successful or not.
+	 * @return bool
+	 */
+	public static function run_setup() {
+		$steps = array(
+			'is_wp_content_writable',
+			'create_advanced_cache',
+			'add_wp_cache_define',
+		);
+
+		foreach ( $steps as $step ) {
+			$result = self::$step();
+			if ( is_wp_error( $result ) ) {
+				jetpack_boost_ds_set( 'page_cache_error', $result->get_error_code() );
+
+				return false;
+			}
+		}
+
+		jetpack_boost_ds_set( 'page_cache_error', '' );
+
+		return true;
+	}
+
+	/**
+	 * Returns true if the wp-content directory is writeable.
+	 */
+	private static function is_wp_content_writable() {
```

Comment:
Minor nitpick: IMHO `is_` functions should only ever return `true` or `false` and should be expected to be used any time you want to find out something about a system.

Perhaps call it `verify_wp_content_writable` or `test_wp_content_writable`. Not a blocker to this PR.

Reply:
Thanks! I've updated this to `verify_wp_content_writable`. I guess the confusion for me comes from returning `true` - thus I added the prefix `is_`.